src/EnsembleTransformKalmanInversion.jl

```diff
@@ -0,0 +1,137 @@
+#Ensemble Transform Kalman Inversion: specific structures and function definitions
+
+"""
+    TransformInversion <: Process
+
+An ensemble transform Kalman Inversion process
+"""
+struct TransformInversion{FT <: AbstractFloat} <: Process
+    # Inverse of the observation error covariance matrix
+    Γ_inv::Union{AbstractMatrix{FT}, UniformScaling{FT}}
+end
+
+function FailureHandler(process::TransformInversion, method::IgnoreFailures)
+    failsafe_update(ekp, u, g, y, obs_noise_cov, failed_ens) = etki_update(ekp, u, g, y, obs_noise_cov)
+    return FailureHandler{TransformInversion, IgnoreFailures}(failsafe_update)
+end
+
+"""
+    FailureHandler(process::TransformInversion, method::SampleSuccGauss)
+
+Provides a failsafe update that
+ - updates the successful ensemble according to the ETKI update,
+ - updates the failed ensemble by sampling from the updated successful ensemble.
+"""
+function FailureHandler(process::TransformInversion, method::SampleSuccGauss)
+    function failsafe_update(ekp, u, g, y, obs_noise_cov, failed_ens)
+        successful_ens = filter(x -> !(x in failed_ens), collect(1:size(g, 2)))
+        n_failed = length(failed_ens)
+        u[:, successful_ens] = etki_update(ekp, u[:, successful_ens], g[:, successful_ens], y, obs_noise_cov)
+        if !isempty(failed_ens)
+            u[:, failed_ens] = sample_empirical_gaussian(u[:, successful_ens], n_failed)
+        end
+        return u
+    end
+    return FailureHandler{TransformInversion, SampleSuccGauss}(failsafe_update)
+end
+
+"""
+     etki_update(
+        ekp::EnsembleKalmanProcess{FT, IT, TransformInversion},
+        u::AbstractMatrix{FT},
+        g::AbstractMatrix{FT},
+        y::AbstractVector{FT},
+        obs_noise_cov::Union{AbstractMatrix{CT}, UniformScaling{CT}},
+    ) where {FT <: Real, IT, CT <: Real}
+
+Returns the updated parameter vectors given their current values and
+the corresponding forward model evaluations.
+"""
+function etki_update(
+    ekp::EnsembleKalmanProcess{FT, IT, TransformInversion{FT}},
+    u::AbstractMatrix{FT},
+    g::AbstractMatrix{FT},
+    y::AbstractVector{FT},
+    obs_noise_cov::Union{AbstractMatrix{CT}, UniformScaling{CT}},
+) where {FT <: Real, IT, CT <: Real}
+
+    m = size(u, 2)
+    Γ_inv = ekp.process.Γ_inv
+    # N_obs × N_obs \ [N_obs × N_ens]
+    # --> tmp is [N_obs × N_ens]
```

Comment:
I assume this comment is an artifact from somewhere? as you don't create tmp here